### PR TITLE
[#5867 : ConceptEntry] Pytorch: HandlingBatches (docs)

### LINKED ISSUE #5867: [Concept Entry] PyTorch: Handling Batches
### Reason/inspiration (optional)

We would like a new entry on `Handling Batches` concept under PyTorch. The entry should go in a new file under `docs/content/pytorch/concepts/handling-batches/handling-batches.md`.       
  
The entry should include:      
  
- An introduction to the concept      
- A `Syntax` section that provides the syntax for the concept (if applicable)      
- An `Example` section that provides an example demonstrating the concept in use (if applicable)     

 
Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.|

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by Swordfish189**
### Description

Added the "Pytorch: Handling Batches" concept as a new entry, via the file docs/content/pytorch/concepts/handling-batches/handling-batches.md

### Issue Solved

Closes #5867

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/pytorch/concepts/handling-batches/handling-batches.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+## Example
+
+Here is an example that demonstrates how to iterate through batches using `DataLoader`:
+
+```py
+for batch_idx, (inputs, labels) in enumerate(dataloader):
+    print(f"Batch {batch_idx + 1}")
+    print("Inputs:", inputs)
+    print("Labels:", labels)
```
Comment (avdhoottt) [inline content/pytorch/concepts/handling-batches/handling-batches.md:58]: Could you please provide the entire example, including how to set up the `DataLoader`? Choose a different example than what you previously demonstrated in the syntax.

File: content/pytorch/concepts/handling-batches/handling-batches.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+```
+
+### Key Parameters
+- **`batch_size`**: The number of samples in each batch.
+- **`shuffle`**: Randomizes the order of data each epoch, ensuring better model generalization.
+- **`num_workers`** (optional): Specifies the number of subprocesses for data loading to speed up training.
+
+## Example
+
+Here is an example that demonstrates how to iterate through batches using `DataLoader`:
```
Comment (avdhoottt) [inline content/pytorch/concepts/handling-batches/handling-batches.md:52]: Add 1-2 lines of description of what the code does.

File: content/pytorch/concepts/handling-batches/handling-batches.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+```py
+for batch_idx, (inputs, labels) in enumerate(dataloader):
+    print(f"Batch {batch_idx + 1}")
+    print("Inputs:", inputs)
+    print("Labels:", labels)
+```
+
+### Expected Output
```
Comment (avdhoottt) [inline content/pytorch/concepts/handling-batches/handling-batches.md:61]: Instead of the heading, simply add a similar line saying, `The output of the above code will be:`.

File: content/pytorch/concepts/handling-batches/handling-batches.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+Inputs: tensor([[3, 4],
+                [1, 2]])
+Labels: tensor([1, 0])
+
+Batch 2
+Inputs: tensor([[5, 6]])
+Labels: tensor([0])
+```
+
+> **Note:** The output order may vary due to `shuffle=True`.
```
Comment (avdhoottt) [inline content/pytorch/concepts/handling-batches/handling-batches.md:74]: Also, include what the `shuffle` parameter does.

File: content/pytorch/concepts/handling-batches/handling-batches.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+### Explanation
```
Comment (avdhoottt) [inline content/pytorch/concepts/handling-batches/handling-batches.md:76]: We don't need to add this sub-heading too.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Hey, I suggested some changes in the entry. Please make them as soon as possible. Thank you!!

**Comment by Swordfish189** [comment]
Linked an unrelated Issue by mistake so I force pushed to correct the mistake. 
Otherwise addressed the requested changes, so should be ready for the next review. :)
Thanks in advance!

File: content/pytorch/concepts/handling-batches/handling-batches.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+  - 'PyTorch'
+  - 'Batch Processing'
+  - 'DataLoader'
+  - 'Neural Networks'
+CatalogContent:
+  - 'intro-to-py-torch-and-neural-networks'
+  - 'paths/build-a-machine-learning-model'
+---
+
+**Handling batches** is an essential practice in **PyTorch** for managing and processing large datasets efficiently. PyTorch simplifies batch handling through the **DataLoader** class. Batch processing groups data samples into fixed-sized subsets, enabling parallel computation, faster training, and better use of GPU resources. This technique is especially critical for deep learning workflows, where training on entire datasets at once is often computationally infeasible.
```
Comment (avdhoottt) [inline content/pytorch/concepts/handling-batches/handling-batches.md:17]:
```suggestion
**Handling batches** is an essential practice in **PyTorch** for managing and processing large datasets efficiently. PyTorch simplifies batch handling through the `DataLoader` class. Batch processing groups data samples into fixed-sized subsets, enabling parallel computation, faster training, and better use of GPU resources. This technique is especially critical for deep learning workflows, where training on entire datasets at once is often computationally infeasible.

```

File: content/pytorch/concepts/handling-batches/handling-batches.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  - 'intro-to-py-torch-and-neural-networks'
+  - 'paths/build-a-machine-learning-model'
+---
+
+**Handling batches** is an essential practice in **PyTorch** for managing and processing large datasets efficiently. PyTorch simplifies batch handling through the `DataLoader` class. Batch processing groups data samples into fixed-sized subsets, enabling parallel computation, faster training, and better use of GPU resources. This technique is especially critical for deep learning workflows, where training on entire datasets at once is often computationally infeasible.
+
+
+## Syntax
+
+PyTorch simplifies batch handling through the `DataLoader` class. Below is the general syntax for setting up a `DataLoader`:
```
Comment (avdhoottt) [inline content/pytorch/concepts/handling-batches/handling-batches.md:22]:
```suggestion
Below is the general syntax for setting up a `DataLoader`:

```

File: content/pytorch/concepts/handling-batches/handling-batches.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+        return len(self.data)
+
+    def __getitem__(self, idx):
+        return self.data[idx], self.labels[idx]
+
+# Instantiate the dataset
+dataset = CustomDataset(data=[[1, 2], [3, 4], [5, 6]], labels=[0, 1, 0])
+
+# Create a DataLoader for batch processing
+dataloader = DataLoader(dataset, batch_size=2, shuffle=True)
```
Comment (avdhoottt) [inline content/pytorch/concepts/handling-batches/handling-batches.md:44]:
```suggestion
from torch.utils.data import DataLoader, Dataset

# Define a custom dataset
class CustomDataset(Dataset):
  def __init__(self, data, labels):
    self.data = data
    self.labels = labels

  def __len__(self):
    return len(self.data)

  def __getitem__(self, idx):
    return self.data[idx], self.labels[idx]

# Instantiate the dataset
dataset = CustomDataset(data=[[1, 2], [3, 4], [5, 6]], labels=[0, 1, 0])

# Create a DataLoader for batch processing
dataloader = DataLoader(dataset, batch_size=2, shuffle=True)

```

File: content/pytorch/concepts/handling-batches/handling-batches.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+```
+
+- **`batch_size`**: The number of samples in each batch.
+- **`shuffle`**: Randomizes the order of data each epoch, ensuring better model generalization.
+- **`num_workers`** (optional): Specifies the number of subprocesses for data loading to speed up training.
+
+## Example
+
+Here is an example that demonstrates how to iterate through batches using DataLoader. This code defines a dataset, initializes the DataLoader, and processes the data in batches:
```
Comment (avdhoottt) [inline content/pytorch/concepts/handling-batches/handling-batches.md:54]:
```suggestion
Here is an example that demonstrates how to iterate through batches using `DataLoader`. This code defines a dataset, initializes the DataLoader, and processes the data in batches:

```

File: content/pytorch/concepts/handling-batches/handling-batches.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+# Create dataset and DataLoader
+dataset = WeatherDataset()
+dataloader = DataLoader(dataset, batch_size=3, shuffle=True)
+
+# Process batches
+for batch_idx, batch in enumerate(dataloader):
+    print(f"Batch {batch_idx + 1}")
+    print("Temperatures:", [temp.item() for temp in batch[0]])
+    print("Weather (1=sunny, 0=cloudy):", [w.item() for w in batch[1]])
+    print()
```
Comment (avdhoottt) [inline content/pytorch/concepts/handling-batches/handling-batches.md:84]:
```suggestion
from torch.utils.data import Dataset, DataLoader

# Define a custom dataset with temperature readings and weather conditions
class WeatherDataset(Dataset):
  def __init__(self):
    # Simple weather data: (temperature, is_sunny)
    self.data = [
      (20, 1), (25, 1), (18, 0), (22, 1),
      (17, 0), (23, 1), (19, 0), (21, 1)
    ]

  def __len__(self):
    return len(self.data)

  def __getitem__(self, idx):
    return self.data[idx]

# Create dataset and DataLoader
dataset = WeatherDataset()
dataloader = DataLoader(dataset, batch_size=3, shuffle=True)

# Process batches
for batch_idx, batch in enumerate(dataloader):
  print(f"Batch {batch_idx + 1}")
  print("Temperatures:", [temp.item() for temp in batch[0]])
  print("Weather (1=sunny, 0=cloudy):", [w.item() for w in batch[1]])
  print()

```

File: content/pytorch/concepts/handling-batches/handling-batches.md [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+dataset = WeatherDataset()
+dataloader = DataLoader(dataset, batch_size=3, shuffle=True)
+
+# Process batches
+for batch_idx, batch in enumerate(dataloader):
+  print(f"Batch {batch_idx + 1}")
+  print("Temperatures:", [temp.item() for temp in batch[0]])
+  print("Weather (1=sunny, 0=cloudy):", [w.item() for w in batch[1]])
+  print()
+
```
Comment (avdhoottt) [inline content/pytorch/concepts/handling-batches/handling-batches.md:85]:
```suggestion
```

File: content/pytorch/concepts/handling-batches/handling-batches.md [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+  def __getitem__(self, idx):
+    return self.data[idx], self.labels[idx]
+
+# Instantiate the dataset
+dataset = CustomDataset(data=[[1, 2], [3, 4], [5, 6]], labels=[0, 1, 0])
+
+# Create a DataLoader for batch processing
+dataloader = DataLoader(dataset, batch_size=2, shuffle=True)
+
```
Comment (avdhoottt) [inline content/pytorch/concepts/handling-batches/handling-batches.md:45]:
```suggestion
```

File: content/pytorch/concepts/handling-batches/handling-batches.md [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+  - 'paths/build-a-machine-learning-model'
+---
+
+**Handling batches** is an essential practice in **PyTorch** for managing and processing large datasets efficiently. PyTorch simplifies batch handling through the `DataLoader` class. Batch processing groups data samples into fixed-sized subsets, enabling parallel computation, faster training, and better use of GPU resources. This technique is especially critical for deep learning workflows, where training on entire datasets at once is often computationally infeasible.
+
+
+## Syntax
+
+Below is the general syntax for setting up a `DataLoader`:
+
```
Comment (avdhoottt) [inline content/pytorch/concepts/handling-batches/handling-batches.md:23]:
```suggestion
```

File: content/pytorch/concepts/handling-batches/handling-batches.md [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+  - 'Batch Processing'
+  - 'DataLoader'
+  - 'Neural Networks'
+CatalogContent:
+  - 'intro-to-py-torch-and-neural-networks'
+  - 'paths/build-a-machine-learning-model'
+---
+
+**Handling batches** is an essential practice in **PyTorch** for managing and processing large datasets efficiently. PyTorch simplifies batch handling through the `DataLoader` class. Batch processing groups data samples into fixed-sized subsets, enabling parallel computation, faster training, and better use of GPU resources. This technique is especially critical for deep learning workflows, where training on entire datasets at once is often computationally infeasible.
+
```
Comment (avdhoottt) [inline content/pytorch/concepts/handling-batches/handling-batches.md:18]:
```suggestion
```

File: content/pytorch/concepts/handling-batches/handling-batches.md [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+```
+
+- **`batch_size`**: The number of samples in each batch.
+- **`shuffle`**: Randomizes the order of data each epoch, ensuring better model generalization.
+- **`num_workers`** (optional): Specifies the number of subprocesses for data loading to speed up training.
+
+## Example
+
+Here is an example that demonstrates how to iterate through batches using `DataLoader`. This code defines a dataset, initializes the DataLoader, and processes the data in batches:
+
```
Comment (avdhoottt) [inline content/pytorch/concepts/handling-batches/handling-batches.md:52]:
```suggestion
```

**Comment by avdhoottt** [comment]
Hey @Swordfish189, thanks for making the changes. This PR is ready for the second review!!

**Review by avdhoottt — APPROVED** [review]

**Review by Sriparno08 — APPROVED** [review]
Looks good, @Swordfish189!
